src/webgpu/shader/execution/expression/call/builtin/faceForward.cache.ts

```diff
@@ -12,30 +12,33 @@ import { IntervalFilter } from '../../interval_filter.js';
 // (vec, vec, vec) -> vec
 //
 // Additionally faceForward has significant complexities around it due to the
-// fact that `dot` is calculated in it s operation, but the result of dot isn't
+// fact that `dot` is calculated in its operation, but the result of dot isn't
 // used to calculate the builtin's result.
 
 /**
  * @returns a Case for `faceForward`
- * @param kind what kind of floating point numbers being operated on
+ * @param data_kind what kind of floating point numbers being operated on
+ * @param op_kind what kind of floating point operation should be performed,
+ *                should be the same as data_kind, except for abstract
  * @param x the `x` param for the case
  * @param y the `y` param for the case
  * @param z the `z` param for the case
  * @param check what interval checking to apply
  * */
 function makeCase(
-  kind: FPKind,
+  data_kind: FPKind,
+  op_kind: FPKind,
```

Comment:
Suggest using the terms `parameter` and `argument`. Also camelCase, not snake_case.
```suggestion
  argumentKind: FPKind,
  parameterKind: FPKind,
```
Here any elsewhere.